### PR TITLE
Added `yumrepo_enabled` parameter

### PR DESCRIPTION
**Pull request by Tamerz**
We want to have the YUM repo available, but we don't want it enabled by default. This is so a simple yum update doesn't upgrade our Wazuh agent since we pin ours to a specific version.

To do this, I added a `yumrepo_enabled` parameter to the `wazuh::agent` and `wazuh::manager` classes.

It will default to true to not break any current people's configuration.

This replaces the previous pull request I put in which I will close: https://github.com/wazuh/wazuh-puppet/pull/479

### CODE REVIEW

**Comment by Tamerz** [comment]
Hello. Can someone please take a look at this when they have a chance?

File: manifests/manager.pp [hunk 1]
```diff
@@ -401,7 +401,8 @@
   # Install and configure Wazuh-manager package
 
   package { $wazuh::params_manager::server_package:
-    ensure  => $server_package_version, # lint:ignore:security_package_pinned_version
+    ensure          => $server_package_version, # lint:ignore:security_package_pinned_version
+    install_options => [ { '--enablerepo'  => 'wazuh' }, ],
```
Comment (c-bordon) [inline manifests/manager.pp:405]: This parameter breaks the installation for Debian or Ubuntu systems:

E: The command line option --enablerepo=wazuh is not understood in combination with the other options

We need that this parameter is not added to these systems

File: manifests/agent.pp [hunk 2]
```diff
@@ -275,15 +276,18 @@
...
         if $::osfamily == 'Debian' {
           Class['wazuh::repo'] -> Class['apt::update'] -> Package[$agent_package_name]
         } else {
           Class['wazuh::repo'] -> Package[$agent_package_name]
         }
       }
       package { $agent_package_name:
-        ensure => "${agent_package_version}-${agent_package_revision}", # lint:ignore:security_package_pinned_version
+        ensure          => "${agent_package_version}-${agent_package_revision}", # lint:ignore:security_package_pinned_version
+        install_options => [{ '--enablerepo'  => 'wazuh' }],
```
Comment (c-bordon) [inline manifests/agent.pp:290]: This parameter breaks the installation for Debian or Ubuntu systems:

E: The command line option --enablerepo=wazuh is not understood in combination with the other options

We need that this parameter is not added to these systems

**Review by c-bordon — CHANGES_REQUESTED** [review]
I was testing the proposed changes and I ran into problems deploying the manager and agent on Debian and Ubuntu systems, so I ask you if you can review the code and add a validation so that this parameter is not included in these systems.

**Comment by Tamerz** [comment]
Honestly, without working unit tests, it is extremely difficult to contribute to this module. I'm going to close this pull request.

**Pull request closed without merging**